server/app/db/crud/bot_types.py

```diff
@@ -0,0 +1,206 @@
+    con: Cursor,
+) -> Tuple[List[bot_type_schema.BotTypeInDB], Union[Exception, None]]:
+    """Fetches all the bots from the database"""
+    logging.info("Trying to fetch all bots")
+
+    # creating query
+    q = MySQLQuery.from_(bot_types).select("*")
+
+    log_query(str(q))
+
+    try:
+        await con.execute(str(q))
+
+        rows = await con.fetchall()
+        # TODO: This might throw an error, need to handle it
+        resp = [bot_type_schema.create_BotTypeInDB_from_tuple(row) for row in rows]
+        logging.debug(
+            f"Got the response:{[x.dict() for x in resp]} when fetching queries"
+        )
+        logging.info("Successfully fetched all the bot_types data")
+        return resp, None
+
+    except Exception as e:
+        logging.error(f"Couldn't query all bot_types with the query={q} due to `{e}`")
+        return [], e
+
+
+async def get_bot_type_with_given_name(
+    con: Cursor, name: str
+) -> Tuple[Union[bot_type_schema.BotTypeInDB, None], Union[Exception, None]]:
+    """Fetch bot with given name"""
+
+    logging.info(f"Trying the fetch bot_type with {name=}")
+
+    # creating query
+    q = MySQLQuery.from_(bot_types).select("*").where(bot_types.name == name)
+
+    log_query(str(q))
+
+    try:
+        await con.execute(str(q))
+        row = await con.fetchone()
+        # TODO: Don't know if this will throw an error or return a empty
+        # response, need to check it
+        if len(row) == 0:
+            return None, RecordNotFound
+        resp = bot_type_schema.create_BotTypeInDB_from_tuple(row)
+        logging.debug(f"for the data={resp.dict()} while fetching bot with {name=}")
+        logging.info(f"Successfully fetched the bot with {name=}")
+        return resp, None
+    except Exception as e:
+        logging.error(f"Unable to query with {name=} with query={q} due to `{e}`")
+        return None, e
+
+
+async def get_bot_type_with_given_id(
+    con: Cursor, id: int
+) -> Tuple[Union[bot_type_schema.BotTypeInDB, None], Union[Exception, None]]:
+    """Fetches the bot with the given id"""
+    logging.info(f"Trying the fetch bot_type with {id=}")
+
+    # creating query
+    q = MySQLQuery.from_(bot_types).select("*").where(bot_types.id == id)
+
+    log_query(str(q))
+
+    # executing the query
+    try:
+        await con.execute(str(q))
+        row = await con.fetchone()
+        # TODO: Don't know if this will throw an error or return a empty
+        # response, need to check it
+        if len(row) == 0:
+            return None, RecordNotFound
+        resp = bot_type_schema.create_BotTypeInDB_from_tuple(row)
+        logging.debug(f"for the data={resp.dict()} while fetching bot with {id=}")
+        logging.info(f"Successfully fetched the bot with the {id=}")
+        return resp, None
+    except Exception as e:
+        logging.error(f"Unable to query with {id=} with query={q} due to `{e}`")
+        return None, e
+
+
+async def update_bot_name(
```

Comment:
Change this to update_botType_name maybe? Thought it changes name in the bots table